rust/kernel/pm_runtime.rs

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Device run-time power management helper functions.
+//!
+//! C header: [`include/linux/pm_runtime.h`](../../../../include/linux/pm_runtime.h)
+
+use crate::{
+    bindings,
+    device::{self, RawDevice},
+    error::Result,
+    to_result,
+};
+
+/// Enable runtime PM of device `dev`.
+pub fn enable(dev: &impl RawDevice) -> DisableGuard {
+    // SAFETY: `dev` is valid by the type invariant.
+    unsafe { bindings::pm_runtime_enable(dev.raw_device()) };
+    DisableGuard::new(device::Device::from_dev(dev))
+}
+
+/// Guard that disables runtime PM of device `dev`.
+pub struct DisableGuard {
+    dev: device::Device,
+}
+
+impl DisableGuard {
+    /// Create new instance of a guard of device `dev`.
+    pub fn new(dev: device::Device) -> Self {
```

Comment:
Is it allowed to call `pm_runtime_disable` without first calling `pm_runtime_enable`?

Reply:
It's not. I see what you mean :)
Thanks.